### DISCUSSION TITLE
basic-audiokit-a2dp.ino ERROR?

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [pschatzmann/arduino-audio-tools#1464] [Q&A] [answered]
hello,
Playing with this Audiokit sketch "basic-audiokit-a2dp.ino" ,i found nothing happened if: 
auto cfg = i2s.defaultConfig(RX_MODE);
But works with:
auto cfg = i2s.defaultConfig(RXTX_MODE);

I recalled an old sketch of about 2 years ago and it was RXTX

Also i cannot  see today's ,an  sketch that shows Audiokit->SD player -> A2DP
Maybe i'm dreaming...

**Comment by pschatzmann** [comment, marked as answer]
Are you sure that you use an actual version? 
I remember having this already [corrected quite recently](https://github.com/pschatzmann/arduino-audio-tools/commit/8acc2bd6ca86474197284f69dc2265fa5d91db61).

**Comment by TangerineDreamer** [comment]
My last  audiotools git clone was the 08 march 2024,nother library involved?

**Comment by pschatzmann** [comment]
As you can see, the correction was on the 12 of March..

**Comment by TangerineDreamer** [comment]
Oh yeah,thanks for the precision,but on the github site the sketch shows : auto cfg = i2s.defaultConfig(RX_MODE);
does it mean the correction you made affects the way of the MODE definition in the sketches now?

**Comment by TangerineDreamer** [comment]
It's working now with auto cfg = i2s.defaultConfig(RX_MODE);
Thanks

**Comment by pschatzmann** [comment]
In the I2S configuration the pin_data  is used in TX_MODE for output and in RX_MODE for input. 
For RXTX_MODE, I am using the pin_data for output and pin_data_rx for input.

This correction makes sure that for the audio kit this logic is applied. Before I was using pin_data_rx for RX_MODE, but this does not give any sound....

**Comment by TangerineDreamer** [comment]
that makes sense :)
